x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { EsArchiver } from 'src/es_archiver';
+import { AppSearchService, IEngine } from '../../../../services/app_search_service';
+import { Browser } from '../../../../../../../test/functional/services/browser';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function enterpriseSearchSetupEnginesTests({
+  getService,
+  getPageObjects,
+}: FtrProviderContext) {
+  const esArchiver = getService('esArchiver') as EsArchiver;
+  const browser = getService('browser') as Browser;
+  const retry = getService('retry');
+  const appSearch = getService('appSearch') as AppSearchService;
+
+  const PageObjects = getPageObjects(['appSearch', 'security']);
+
+  describe('Engines Overview', function() {
+    let engine1: IEngine;
+    let engine2: IEngine;
+    let metaEngine: IEngine;
+
+    before(async () => {
+      await esArchiver.load('empty_kibana');
+      engine1 = await appSearch.createEngine();
+      engine2 = await appSearch.createEngine();
+      metaEngine = await appSearch.createMetaEngine([engine1.name, engine2.name]);
+    });
+
+    after(async () => {
+      await esArchiver.unload('empty_kibana');
+      appSearch.destroyEngine(engine1.name);
+      appSearch.destroyEngine(engine2.name);
+      appSearch.destroyEngine(metaEngine.name);
+    });
+
+    describe('when an enterpriseSearch.host is configured', () => {
+      it('navigating to the enterprise_search plugin will redirect a user to the App Search Engines Overview page', async () => {
+        await PageObjects.security.forceLogout();
+        const { user, password } = appSearch.getEnterpriseSearchUser();
+        await PageObjects.security.login(user, password, {
+          expectSpaceSelector: false,
+        });
+
+        await PageObjects.appSearch.navigateToPage();
+        await retry.try(async function() {
+          const currentUrl = await browser.getCurrentUrl();
+          expect(currentUrl).to.contain('/app_search');
+        });
+      });
+
+      it('lists engines', async () => {
+        const engineLinks = await PageObjects.appSearch.getEngineLinks();
+        const engineLinksText = await Promise.all(engineLinks.map(l => l.getVisibleText()));
```

Comment:
Super interesting syntax - definitely very different from Cypress, that's for sure!

Reply:
There may be a cleaner way in there to do that, I'm not sure. 

The fact that `getVisibleText()` returns a Promise required the Promise.all...  I do wonder if it supports calling `.getVisibleText()` on the collection?

Reply:
I'm not familiar enough with Cypress so I'm not sure what the equivalent would look like.